tests/integration_tests/test_ray.py

```diff
@@ -254,74 +254,62 @@ def test_ray_read_binary_files(tmpdir, df_engine, ray_cluster_2cpu):
     assert proc_col.equals(proc_col_expected)
 
 
-# TODO(geoffrey): Add dataset_type="csv" back to parameters if we can prevent CI timeouts.
-@pytest.mark.parametrize("dataset_type", ["parquet"])
 @pytest.mark.distributed
-def test_ray_save_inputs_with_nans(tmpdir, dataset_type, ray_cluster_2cpu):
-    image_dest_folder = os.path.join(tmpdir, "generated_images")
-    audio_dest_folder = os.path.join(tmpdir, "generated_audio")
-    input_features = [
-        image_feature(
-            folder=image_dest_folder,
-            preprocessing={"in_memory": True, "height": 12, "width": 12, "num_channels": 3, "num_processes": 5},
-            encoder={"output_size": 16, "num_filters": 8},
-        ),
-        audio_feature(
-            folder=audio_dest_folder,
-            preprocessing={
-                "audio_file_length_limit_in_s": 3.0,
-                "missing_value_strategy": BFILL,
-                "in_memory": True,
-                "padding_value": 0,
-                "norm": "per_file",
-                "type": "fbank",
-                "window_length_in_s": 0.04,
-                "window_shift_in_s": 0.02,
-                "num_filter_bands": 80,
-            },
+@pytest.mark.parametrize(
+    "df_engine",
+    [
+        "dask",
+        pytest.param(
+            "modin",
+            marks=pytest.mark.skipif(_modin_ray_incompatible, reason="modin<=0.15.2 does not support ray>=1.13.0"),
         ),
-        sequence_feature(encoder={"reduce_output": "sum"}),
+    ],
+)
+def test_ray_tabular(df_engine, ray_cluster_2cpu):
+    input_features = [
         category_feature(encoder={"vocab_size": 2}, reduce_input="sum"),
         number_feature(normalization="zscore"),
         set_feature(),
         binary_feature(),
         bag_feature(),
-        text_feature(),
-        timeseries_feature(),
+        h3_feature(),
         date_feature(),
-        # TODO: NaN handling not supported. See `test_ray_save_inputs_and_outputs_without_nans` below.
-        # vector_feature(),  # NaNs are not supported by the feature
-        # TODO: feature type not yet supported
-        # h3_feature(),  # ValueError casting large int strings (e.g. '5.864041857092157e+17') to int: MLI-72
     ]
     output_features = [
-        category_feature(decoder={"vocab_size": 5}),  # Regression test for #1991 requires multi-class predictions.
+        binary_feature(bool2str=["No", "Yes"]),
+        binary_feature(),
```

Comment:
Just curious - do we intentionally have 2 separate binary features as output features for this test?

Reply:
Good question! This has always been there– presumably it was initially a test for the `bool2str` kwarg– so I left it as-is

Reply:
Got it, interesting that we have both, wish there was an explanation for why we're doing it when this test was added. Should be fine for now

Reply:
At one point in time, there was a bug in Ray Datasets that was casting these string bools into objects:

https://github.com/ludwig-ai/ludwig/pull/1812